analyses/sex-prediction-from-RNASeq/sex-prediction-from-RNASeq.Rmd

```diff
@@ -0,0 +1,168 @@
+---
+title: "R Notebook"
+output: html_notebook
+---
+
+RNA-Seq data is in pbta-gene-expression-kallisto.rds.
+
+```{r}
+
+ge <- readRDS("../../data/pbta-gene-expression-kallisto.rds")
+
+
+```
+
+Metadata is in pbta-histologies.tsv.  reported_gender is our target variable. Kids_First_Biospecimen_ID is the unique identifier for each sample. 
+
+```{r}
+hist <- read.delim("../../data/pbta-histologies.tsv", header=TRUE, sep="\t", stringsAsFactors = FALSE)
+
+missing_reported_gender_samples <- hist[which(hist[, "reported_gender"] == ""), "Kids_First_Biospecimen_ID"]
+
+```
+
+Calculate the median absolute deviation for each transcript.
+
+```{r}
+library(stats)
+
+mads <- apply(ge[ , 3:1030], 1, function(x) mad(x, high=TRUE))
+
+mads_lo_hi <- sort(mads, index.return=TRUE)
+
+```
+
+Let's try a model with the top 25% highest mads.
+Start a df with just the rows of ge corresponding to the top 25% highest mads.  Call it ge_top.
+transpopse ge_top[, 3:1030]. Call it df.
+Set colnames(df) to transcript IDs.
+Set rownames(df) to sample names.
+
+```{r}
+tail_percent = 0.75
+ge_top <- ge[mads_lo_hi$ix[floor(tail_percent*length(mads)):length(mads)], ]
+df <- t(ge_top[, 3:1030])
+
+colnames(df) <- ge_top[, 1]
+
+rownames(df) <- colnames(ge_top[, 3:1030])
+
+
+```
+
+Eliminate rows of df corresponding to samples with missing reported_gender.
+```{r}
+
+df <- df[!rownames(df) %in% missing_reported_gender_samples, ]
+
+```
+
+Use reported_gender column from hist as the response.
+
+Extract reported_gender values from reported_gender column of hist.  Put these value in a two-column dataframe
+  c("Kids_First_Biospecimen_ID", "reported_gender").
+
+```{r}
+
+reported_gender <- hist[hist$Kids_First_Biospecimen_ID %in% rownames(df), c("Kids_First_Biospecimen_ID", "reported_gender")]
+
+```
+
+Check sequence of rownames(df) and reported_gender[, 1].  reported_gender_response holds the reported_gender values in the same Kids_First_Biospecimen_ID sequence as rownames(df).
+
+```{r}
+
+match_index <- unlist(sapply(rownames(df), function(x) which(reported_gender[, 1] == x)))
+
+reported_gender_response <- reported_gender[match_index, 2]
+```
+
+Some values in reported_gender_response = "Not Available".  Eliminate from both reported_gender_response and df.
+
+```{r}
+reported_gender_response_hold <- reported_gender_response
+reported_gender_response <- reported_gender_response[reported_gender_response != "Not Available"]
+df <- df[reported_gender_response_hold != "Not Available", ]
+```
+
+
+Build predictive model for reported_gender_response.
+Hold out 30% of the samples to test accuracy after cross-validation tuning of 
+  lambda and alpha regularization parameters.
+
+```{r}
+
+library(glmnet)
```

Comment:
Can you put all the `library` calls in a single chunk up at the top of the file please?